src/seedu/addressbook/common/Utils.java

```diff
@@ -64,4 +67,24 @@ public static boolean isValidDate(String value) {
         }
         return isValid;
     }
+
+    /**
+     * Custom comparator for Date string in Fees in the form of DD-MM-YYYY
+     * Allows for sorting of Person's list according to YYYYMMDD of Fees section.
+     */
+    public static class FeesComparator implements Comparator<Person> {
+        @Override
+        public int compare(Person o1, Person o2) {
+            StringBuilder main1 = new StringBuilder();
+            StringBuilder main2 = new StringBuilder();
+            main1.append(o1.getFees().duedate.substring(6, 10));
+            main1.append(o1.getFees().duedate.substring(3, 5));
+            main1.append(o1.getFees().duedate.substring(0, 2));
+            main2.append(o2.getFees().duedate.substring(6, 10));
+            main2.append(o2.getFees().duedate.substring(3, 5));
+            main2.append(o2.getFees().duedate.substring(0, 2));
+            return main1.toString().compareTo
+                    (main2.toString());
+        }
+    }
```

Comment:
Something is wrong when even I don't understand this code.
Also, duedate should not be public

Reply:
u mean the comparator or this particular return thing?

Reply:
what does all the append/substring mean?

Reply:
its removing all the dashes in the DD-MM-YYYY and appending into a string for YYYYMMDD to sort

Reply:
So it'll be good to pull it into its own function called convertToString or something

Reply:
Ok I'll convert it! Should i put it under Utils also?

Reply:
Then you can call it in your UniquePersonList also

Reply:
Gotcha